distributed/client.py

```diff
@@ -1121,6 +1129,8 @@ def _send_to_scheduler(self, msg):
 
     async def _start(self, timeout=no_default, **kwargs):
         self.status = "connecting"
+        if self._set_as_default:
+            _set_global_client(self)
```

Comment:
I think moving `_set_global_client` into `async def _start` is worse - as this is likely to be called via `Client(asynchronous=False)` and so run in an off-main-thread eventloop thread